CMakeLists.txt

```diff
@@ -126,13 +132,25 @@ else()
     else ()
         set(EMPTYDIR "%empty")
     endif()
+	
     configure_file(${PROJECT_SOURCE_DIR}/src/yang.y.in ${PROJECT_BINARY_DIR}/src/yang.y)
-    add_custom_target(bison
-        WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}/src
-        COMMAND bison -l -o parser_yang_bis.c --defines=parser_yang_bis.h ${PROJECT_BINARY_DIR}/src/yang.y
-        COMMAND flex -8 -L -o parser_yang_lex.c --header-file=parser_yang_lex.h yang.l
-        COMMAND ${SED_TOOL} -i 's/int yychar\;/int yychar\;\\nchar *s = NULL, *tmp_s = NULL, *ext_name = NULL\;\\nstruct lys_module *trg = NULL\;\\nstruct lys_node *tpdf_parent = NULL, *data_node = NULL\;\\nstruct lys_ext_instance_complex *ext_instance = NULL\;\\nint is_ext_instance\;\\nvoid *actual = NULL\;\\nenum yytokentype backup_type, actual_type = MODULE_KEYWORD\;\\nint64_t cnt_val = 0\;\\nint is_value = 0\;\\nvoid *yang_type = NULL\;/' parser_yang_bis.c
-    )
+	
+    if (WIN32)
+	    add_custom_target(bison
+            WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}/src
+            COMMAND ${BISON_EXECUTABLE} -l -o parser_yang_bis.c --defines=parser_yang_bis.h ${PROJECT_BINARY_DIR}/src/yang.y
+            COMMAND ${FLEX_EXECUTABLE} -8 -L -o parser_yang_lex.c --header-file=parser_yang_lex.h yang.l
+            COMMAND ${SED_TOOL} -i "s/int yychar\;/int yychar\;\\nchar *s = NULL, *tmp_s = NULL, *ext_name = NULL\;\\nstruct lys_module *trg = NULL\;\\nstruct lys_node *tpdf_parent = NULL, *data_node = NULL\;\\nstruct lys_ext_instance_complex *ext_instance = NULL\;\\nint is_ext_instance\;\\nvoid *actual = NULL\;\\nenum yytokentype backup_type, actual_type = MODULE_KEYWORD\;\\nint64_t cnt_val = 0\;\\nint is_value = 0\;\\nvoid *yang_type = NULL\;/" parser_yang_bis.c
+        )
+		add_definitions(/DYY_NO_UNISTD_H)
+	else()
+	    add_custom_target(bison
+            WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}/src
+            COMMAND ${BISON_EXECUTABLE} -l -o parser_yang_bis.c --defines=parser_yang_bis.h ${PROJECT_BINARY_DIR}/src/yang.y
+            COMMAND ${FLEX_EXECUTABLE} -8 -L -o parser_yang_lex.c --header-file=parser_yang_lex.h yang.l
+            COMMAND ${SED_TOOL} -i 's/int yychar\;/int yychar\;\\nchar *s = NULL, *tmp_s = NULL, *ext_name = NULL\;\\nstruct lys_module *trg = NULL\;\\nstruct lys_node *tpdf_parent = NULL, *data_node = NULL\;\\nstruct lys_ext_instance_complex *ext_instance = NULL\;\\nint is_ext_instance\;\\nvoid *actual = NULL\;\\nenum yytokentype backup_type, actual_type = MODULE_KEYWORD\;\\nint64_t cnt_val = 0\;\\nint is_value = 0\;\\nvoid *yang_type = NULL\;/' parser_yang_bis.c
+        )
+	endif()	   
```

Comment:
I can see only 2 differences between WIN and LINUX version, the new definition and using `"` instead of `'`. I believe the quote change is not really worth this whole change and I am quite sure it can be unified. Could you please try to do that?

Reply:
Sure. I now use a temporary variable to store quoted sed statements.